python/dgl/distributed/graph_services.py

```diff
@@ -625,28 +660,58 @@ def sample_etype_neighbors(
         nodes = F.cat(homo_nids, 0)
 
     def issue_remote_req(node_ids):
+        if prob is not None:
+            # See NOTE 1
+            _prob = [
+                g.edges[etype].data[prob].kvstore_key
+                if prob in g.edges[etype].data
+                else ""
+                for etype in g.etypes
```

Comment:
let's always use canonical etype if possible.

Reply:
Unfortunately DistGraph's `edges[]` does not accept canonical edge type for now.

Reply:
all right, I will handle this in my PR of canonical etype support